examples/cluster/plot_hierarchical_clustering_dendrogram.py

```diff
@@ -0,0 +1,45 @@
+# Authors: Mathew Kallada
+# License: BSD 3 clause
+"""
+=========================================
+Plot Hierarachical Clustering Dendrogram 
+=========================================
+
+This example plots the corresponding dendrogram of a hierarchical clustering
+using AgglomerativeClustering and the dendrogram method available in scipy.
+"""
+
+import numpy as np
+
+from matplotlib import pyplot as plt
+from scipy.cluster.hierarchy import dendrogram
+from sklearn.datasets import load_iris
+from sklearn.cluster import AgglomerativeClustering
+
+def plot_dendrogram(model, **kwargs):
+
+    # Children of hierarchical clustering
+    children = model.children_
+
+    # Distances between each pair of children
+    # Since we don't have this information, we can use a uniform one for plotting
+    distance = np.arange(children.shape[0])
+
+    # The number of observations contained in each cluster level
+    no_of_observations = np.arange(2, children.shape[0]+2)
+
+    # Create linkage matrix and then plot the dendrogram
+    linkage_matrix = np.column_stack([children, distance, no_of_observations]).astype(float)
+
+    # Plot the corresponding dendrogram
+    dendrogram(linkage_matrix, **kwargs)
+
+
+iris = load_iris()
+x = iris.data[:20]
```

Comment:
actually playing around with this i don't think this is a good idea at all:
```python
In [3]: iris.target[:20]
Out[3]: array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
```

you're just focusing on one class of the iris dataset, i'd just focus on the whole and would want to see if the clustering comes up with 3 reasonable classes...

Also i think an example with lots of samples is more useful as it would also show how to deal with pruning...